src/components/structures/RightPanel.tsx

```diff
@@ -293,21 +249,22 @@ export default class RightPanel extends React.Component<IProps, IState> {
 
             case RightPanelPhases.RoomMemberInfo:
             case RightPanelPhases.SpaceMemberInfo:
-            case RightPanelPhases.EncryptionPanel:
+            case RightPanelPhases.EncryptionPanel: {
+                const roomMember = this.state.member instanceof RoomMember ? this.state.member: undefined;
                 panel = <UserInfo
                     user={this.state.member}
-                    room={this.context.getRoom(this.state.member.roomId) ?? this.props.room}
+                    room={this.context.getRoom(roomMember?.roomId) ?? this.props.room}
                     key={roomId || this.state.member.userId}
                     onClose={this.onClose}
                     phase={this.state.phase}
                     verificationRequest={this.state.verificationRequest}
                     verificationRequestPromise={this.state.verificationRequestPromise}
                 />;
                 break;
-
+            }
             case RightPanelPhases.Room3pidMemberInfo:
             case RightPanelPhases.Space3pidMemberInfo:
-                panel = <ThirdPartyMemberInfo event={this.state.event} key={roomId} />;
+                panel = <ThirdPartyMemberInfo event={this.state.threadHeadEvent} key={roomId} />;
```

Comment:
Why was the state field renamed to `threadHeadEvent`...? It seems to also be used for various non-thread cases, so the name seems confusing.

Reply:
There was lots of confusion between thread head event and `initialEvent` (the highlighted event in a thread). So I wanted the event names to be more specific in the `ICardState` I think the best solution would be to add more fields to the `ICardState`. So It is not confusing for threads, but we dont have threadHeadEvents where it makes no sense.

Reply:
Ah okay, not sure of all the cases you need to cover, but yes perhaps more fields could work. If it's a separate field for every case, then that seems like too much, so hopefully there's a balance we can find.

Reply:
its just the one event for the thirdPartyMember